### PR TITLE
covers all enumerables, passes some tests

### PR DESCRIPTION
**Pull request by danwhitston**
Fails one test which checks if passing an argument for the initial value works. It, um, doesn't. Still trying to figure out why.

### CODE REVIEW

**Comment by danwhitston** [comment]
All done! Ideally I'd tidy up the code and either split the method or find ways of reducing its complexity, but it works with every inject syntax I threw at it, and with both hashes and arrays. I may try and think of something else that's enumerable and give it a go before Monday if I'm feeling super-swish, but I doubt it. Also doubt I'll get round to re-implementing using recursion rather than enumeration.

File: lib/danject.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module Enumerable
+  def danject(*accum)
```
Comment (ptolemybarnes) [inline lib/danject.rb:2]: Why can your inject take unlimited params? core inject can't.

File: lib/danject.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        self.drop(1).each do |item|
+          accum = yield(accum, item)
+        end
+      else
+        accum = accum[0]
+        self.each do |item|
+          accum = yield(accum, item)
+        end
+      end
+      return accum
```
Comment (ptolemybarnes) [inline lib/danject.rb:15]: You needn't explicitly return here and below

**Comment by ptolemybarnes** [comment]
Looks pretty good! But definitely in need of refactoring as there is a great deal of repeated code.

**Comment by silvabox** [comment]
Well done Daniel but it definitely needs a refactor to remove the nested `if`s and the long method.  :checkered_flag:

**Pull request closed without merging**